src/deepforest/main.py

```diff
@@ -434,226 +422,158 @@ def predict_image(self,
-            (deprecated) savedir: directory to save images with bounding boxes
-            (deprecated) color: color of the bounding box as a tuple of BGR color, e.g. orange annotations is (0, 165, 255)
-            (deprecated) thickness: thickness of the rectangle border line in px
-
+            crop_model: a deepforest.model.CropModel object to predict on crops
+            size: the size of the image to resize to. Optional, if not provided, the image is not resized.
         Returns:
             df: pandas dataframe with bounding boxes, label and scores for each image in the csv file
         """
 
-        df = utilities.read_file(csv_file)
-        ds = dataset.TreeDataset(csv_file=csv_file,
-                                 root_dir=root_dir,
-                                 transforms=None,
-                                 train=False)
-        dataloader = self.predict_dataloader(ds)
+        ds = prediction.FromCSVFile(csv_file=csv_file, root_dir=root_dir, size=size)
 
+        dataloader = self.predict_dataloader(ds)
         results = predict._dataloader_wrapper_(model=self,
+                                               crop_model=crop_model,
                                                trainer=self.trainer,
-                                               annotations=df,
                                                dataloader=dataloader,
-                                               root_dir=root_dir,
-                                               nms_thresh=self.config.nms_thresh,
-                                               color=color,
-                                               savedir=savedir,
-                                               thickness=thickness)
+                                               root_dir=root_dir)
 
         results.root_dir = root_dir
 
         return results
 
     def predict_tile(self,
-                     raster_path=None,
                      path=None,
+                     paths=None,
                      image=None,
                      patch_size=400,
                      patch_overlap=0.05,
                      iou_threshold=0.15,
-                     in_memory=True,
-                     return_plot=False,
-                     mosaic=True,
-                     sigma=0.5,
-                     thresh=0.001,
-                     color=None,
-                     thickness=1,
-                     crop_model=None,
-                     crop_transform=None,
-                     crop_augment=False):
+                     dataloader_strategy="single",
+                     crop_model=None):
         """For images too large to input into the model, predict_tile cuts the
         image into overlapping windows, predicts trees on each window and
         reassambles into a single array.
 
         Args:
-            raster_path: [Deprecated] Use 'path' instead
             path: Path to image on disk
+            paths: List of paths to images on disk
```

Comment:
This API feels a little awkward to me. My initial thought is that the argument should just be `paths` (no `path`) and then we should check the length and how it relates to `dataloader_strategy` and report back a useful error if the user is providing more than one path and has chosen a loader that is only appropriate for a single path. Happy to chat about this more if there's good reason for separate arguments.

Also, it seems natural that if we can take multiple paths we'd also be able to take multiple images. I haven't made it into the relevant part of the code yet so not sure if there's something that prevents that.

Reply:
The challenge is whether this should be paths = [<path on disk>]
or paths = <path of disk>

in the case whether is just one path. It seems weird to have a plural argname and NOT have a list? @jveitchmichaelis? @henrykironde?

Reply:
ya, I went to feed the cats and came to same idea. reverting my commit.

Reply:
Sounds good. We should make sure we have tests for all 3 cases: 1) single path as string; 2) single path as one item list; 3) multiple paths